app/code/Magento/StoreGraphQl/Model/Resolver/Store/StoreConfigDataProvider.php

```diff
@@ -55,24 +56,26 @@ public function __construct(
      */
     public function getStoreConfigData(StoreInterface $store): array
     {
-        $storeConfigData = array_merge(
-            $this->getBaseConfigData($store),
-            $this->getExtendedConfigData((int)$store->getId())
-        );
-        return $storeConfigData;
+        $stores = $this->storeConfigManager->getStoreConfigs();
+        $defaultStoreConfig = current($this->storeConfigManager->getStoreConfigs([$store->getCode()]));
+        $defaultStoreConfigData = $this->prepareStoreConfigData($defaultStoreConfig);
+
+        foreach ($stores as $storeConfig) {
+            $defaultStoreConfigData['stores'][] = $this->prepareStoreConfigData($storeConfig);
+        }
+
+        return $defaultStoreConfigData;
     }
 
     /**
-     * Get base config data
+     * Prepare store config data
      *
-     * @param StoreInterface $store
+     * @param StoreConfigInterface $storeConfig
      * @return array
      */
-    private function getBaseConfigData(StoreInterface $store) : array
+    private function prepareStoreConfigData($storeConfig): array
```

Comment:
Can we add `StoreConfigInterface` to type hints in method arguments?

Reply:
Hello @swnsma , the change has been done.